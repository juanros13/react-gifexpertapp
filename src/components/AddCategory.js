import React, { useState } from 'react';
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => {
    const [inputValue, setInputValue] = useState('');

    const hanledInputChange = (e) =>{
        setInputValue(e.target.value);
        // console.log(e.target.value);
    }
    const hanledSubmit = (e) => {
        e.preventDefault();
        // console.log('submit hecho', inputValue);
        if(inputValue.trim().length>2){
            setCategories(cats => [ inputValue,...cats,]);
            setInputValue('');
        }
        
    }

    return (
        <form onSubmit={ hanledSubmit }>
            {/* <h1>{ inputValue }</h1> */}
            <p> { inputValue } </p>
            <input
                type='text'
                value={ inputValue }
                onChange={ hanledInputChange }
                
            />
        </form>
    )
}
AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
