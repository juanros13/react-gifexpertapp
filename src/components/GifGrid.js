import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
// import { getGifs } from '../helpers/getGifs';
import GifGridItem  from './GifGridItem';

import PropTypes from 'prop-types';

export const GifGrid = ( { category } ) => {
    // const [counter, setCounter] = useState(0);
    const { data:images, loading } = useFetchGifs(category);

    // const [images, setImages] = useState([])
    // useEffect(()=>{
    //     getGifs(category)
    //     .then(imgs => setImages(imgs));
    // },[category]);

    //getGifs();
    return (
        <>
        <h3 className='animate__animated animate__fadeIn'>{ category }</h3>
        { loading && <p className='animate__animated animate__flash'>Loading</p>}

        <div className='card-gif'>
            {
                images.map((img) =>
                    // <li  key = { id }>{ title }</li>
                    <GifGridItem
                        key = { img.id }
                        { ...img }
                    />
                )
            }
        </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}