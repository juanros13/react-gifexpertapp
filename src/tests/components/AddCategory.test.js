import React from 'react';
import { shallow } from "enzyme"
import { AddCategory } from "../../components/AddCategory";
import '@testing-library/jest-dom';

describe('Pruebas en el componente AddCategory', () => {

    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories = { setCategories }/>);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories = { setCategories }/>);
    })


    test('Debe mostrarse correctamente AddCategory', () => {
        expect( wrapper ).toMatchSnapshot();
    });
    test('Debe cambiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value =  'Hola mundo';
        input.simulate('change', { target: { value: value } });
        expect(wrapper.find('p').text().trim()).toBe(value + "");
    });

    test('No debe de postear la informacion en submit', () => {
        wrapper.find('form').simulate('submit', { preventDefault(){} } );
        expect( setCategories ).not.toHaveBeenCalled();
    });
    test('debe llamar el setCategory y limpiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value =  'Hola mundo';
        input.simulate('change', { target: { value: value } });
        wrapper.find('form').simulate('submit', { preventDefault(){} } );
        expect( setCategories ).toHaveBeenCalled();
        expect(wrapper.find('input').prop('value') ).toBe('');
    })
    
    
    
    
})
