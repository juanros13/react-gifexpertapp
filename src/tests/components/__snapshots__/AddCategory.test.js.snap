// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente AddCategory Debe mostrarse correctamente AddCategory 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
     
     
  </p>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
